### PR TITLE
Parsing data from zoneinfo

### PR DESCRIPTION
**Pull request by ppalucki**
New set of metrics based on /proc/zoneinfo

Includes fix to be able to specify more patches for runner tests and unit tests for parsing zoneinfo.

image

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -84,7 +84,15 @@ and store them in metrics_storage component.
 
     Attach following labels to all metrics:
     `sockets`, `cores`, `cpus`, `cpu_model`, `cpu_model_number` and `wca_version`
-    (defaults to False)
+
+- ``zoneinfo``: **Union[Str, bool]** = *True*
+
+    True means use the sane default.
```
Comment (mmucek95) [inline docs/api.rst:90]: We wouldn't like to use **insane** default. That would be bad.

File: docs/api.rst [hunk 2]
```diff
@@ -84,7 +84,15 @@ and store them in metrics_storage component.
...
+    False means disable the collection.
+    If string is provided it is regexp to extract information from /proc/zoneinfo
+    (only matching regexp will be collected, with key from frist group and value from
```
Comment (mmucek95) [inline docs/api.rst:93]: s/frist/first

File: tests/test_zoneinfo.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+from wca.metrics import MetricName
+from wca.zoneinfo import get_zoneinfo_measurements, DEFAULT_REGEXP
+
+
+@patch('builtins.open', new=create_open_mock({
+    "/proc/zoneinfo": open(relative_module_path(__file__, 'fixtures/proc-zoneinfo.txt')).read(),
+}))
+def test_parse_proc_zoneinfo(*mocks):
+    assert get_zoneinfo_measurements(re.compile(DEFAULT_REGEXP)) == {
+        MetricName.PLATFORM_ZONEINFO: {'0': {'DMA': {'high': 6.0,
```
Comment (mmucek95) [inline tests/test_zoneinfo.py:27]: Have you considered moving this directory to a fixture?

File: wca/runners/measurement.py [hunk 4]
```diff
@@ -163,7 +165,15 @@ class MeasurementRunner(Runner):
 
         Attach following labels to all metrics:
         `sockets`, `cores`, `cpus`, `cpu_model`, `cpu_model_number` and `wca_version`
-        (defaults to False)
+
+    - ``zoneinfo``: **Union[Str, bool]** = *True*
+
+        True means use the sane default.
```
Comment (mmucek95) [inline wca/runners/measurement.py:171]: *Sane* again. I don't know what is it supposed to mean.

File: wca/runners/measurement.py [hunk 5]
```diff
@@ -163,7 +165,15 @@ class MeasurementRunner(Runner):
...
+        False means disable the collection.
+        If string is provided it is regexp to extract information from /proc/zoneinfo
+        (only matching regexp will be collected, with key from frist group and value from
```
Comment (mmucek95) [inline wca/runners/measurement.py:174]: s/frist/first

File: tests/test_zoneinfo.py (hunk 3 above)
Comment (ppalucki) [reply tests/test_zoneinfo.py:27]: fixture is an input, I do not want to serialize excepted results to be consistent with all other results.

File: wca/runners/measurement.py (hunk 5 above)
Comment (ppalucki) [reply wca/runners/measurement.py:174]: fixed

File: wca/runners/measurement.py (hunk 4 above)
Comment (ppalucki) [reply wca/runners/measurement.py:171]: fixed

**Pull request merged**